### PR TITLE
test(core): test coverage for EvmTransactionResponse

### PR DESCRIPTION
**Pull request by sogunshola**
---
name: 'Pull request'
about: A new pull request
---

### Checklist

- [X] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [X] My code is conform the [code style](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/CODE_STYLE.md)
- [X] I have made corresponding changes to the documentation
- [X] I have updated Typescript definitions when needed

### Issue Description

Related issue: #`FILL_THIS_OUT`

### CODE REVIEW

File: packages/core/src/test/EvmTransactionResponse.test.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+// Set variables
```
Comment (b4rtaz) [inline packages/core/src/test/EvmTransactionResponse.test.ts:1]: Should this comment be below the import section?

File: packages/core/src/test/EvmTransactionResponse.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    const equality = transactionResponseA.equals(transactionResponseB);
+
+    expect(equality).toBe(false);
+  });
+
+  it('should convert EvmTransactionResponse to JSON', () => {
+    const transactionResponse = EvmTransactionResponse.create(inputWithAllData);
+    const result = transactionResponse.toJSON();
+
+    expect(result.to).toBe(inputWithAllData.to);
```
Comment (b4rtaz) [inline packages/core/src/test/EvmTransactionResponse.test.ts:108]: in this place, I think we need more expectations.

File: packages/core/src/test/EvmTransactionResponse.test.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    expect(equality).toBe(true);
+  });
+
+  it('should confirm inequality of 2 EvmTransactionResponses', () => {
+    const transactionResponseA = EvmTransactionResponse.create(inputWithAllData);
+    const transactionResponseB = EvmTransactionResponse.create(requiredParamsOnly);
+    const equality = transactionResponseA.equals(transactionResponseB);
+
+    expect(equality).toBe(false);
+  });
```
Comment (ErnoW) [inline packages/core/src/test/EvmTransactionResponse.test.ts:102]: Can you add more cases for equality checks?
If you run `yarn test:coverage` then you can see line coverage. Some cases are not yet included in the unit tests

**Review by ErnoW — CHANGES_REQUESTED** [review]
Nice work, please add more coverage for the equality tests.
Also the wait function is not covered.

run `yarn test:coverage` in the core package to get a coverage report.

File: packages/evmUtils/src/dataTypes/EvmTransactionResponse/EvmTransactionResponse.test.ts [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+  gasPrice: 21000,
+  gasUsed: 21000,
+  logs: [],
+  status: 1,
+};
+
+describe('EvmTransactionResponse', () => {
+  beforeAll(() => {
+    setupEvmUtils();
+  });
```
Comment (b4rtaz) [inline packages/evmUtils/src/dataTypes/EvmTransactionResponse/EvmTransactionResponse.test.ts:54]: It this necessary?

Comment (sogunshola) [reply packages/evmUtils/src/dataTypes/EvmTransactionResponse/EvmTransactionResponse.test.ts:54]: Yes. the module has to be registered first. That's the case in all tests for evmUtils

Comment (b4rtaz) [reply packages/evmUtils/src/dataTypes/EvmTransactionResponse/EvmTransactionResponse.test.ts:54]: But you don't test the module here. You test just only one class without a reference to modules. If the module setup is not necessary to test the class, you shouldn't call it. Tests should touch only minimal region of the code.

**Review by b4rtaz — APPROVED** [review]

File: packages/evmUtils/src/dataTypes/EvmTransactionResponse/EvmTransactionResponse.test.ts (hunk 4 above)
Comment (b4rtaz) [reply packages/evmUtils/src/dataTypes/EvmTransactionResponse/EvmTransactionResponse.test.ts:54]: After @sogunshola feedback: this setup is necessary.

**Review by b4rtaz — APPROVED** [review]

**Pull request closed without merging**